pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1493,6 +1537,26 @@ func (f *schemaChangeFrontier) SpanFrontier() *span.Frontier {
 	return f.spanFrontier.Frontier
 }
 
+func (f *schemaChangeFrontier) checkpoint(c *jobspb.ChangefeedProgress_Checkpoint) {
+	// Checkpoint record could be fairly large.
+	// Assume we have a 10T table, and a 1/2G max range size: 20K spans.
+	// Span frontier merges adjacent spans, so worst case we have 10K spans.
+	// Each span is a pair of keys.  Those could be large.  Assume 1/2K per key.
+	// So, 1KB per span.  We could be looking at 10MB checkpoint record.
+	// Since we checkpoint infrequently, a 10MB write might be okay.
```

Comment:
I'd love some better control here. 10MiB scares me. It feels to me like adding some knob on the max size would be very valuable. I'd do it just by throwing away later spans. You'll still make progress so long as you get one resolved span from the worker with the lexicographically highest span because it'll be adjacent. How about a cluster setting and then you just stop appending spans once you hit the threshold?